code/game/objects/items/tanks/jetpack.dm

```diff
@@ -67,7 +69,20 @@
 	user.remove_movespeed_modifier(MOVESPEED_ID_JETPACK)
 
 /obj/item/tank/jetpack/proc/move_react(mob/user)
-	allow_thrust(0.01, user)
+	if(!on)//If jet dont work, it dont work
+		return
+	if(!user)//Don't allow jet self using
+		return
+	if(!isturf(user.loc))//You can't use jet in nowhere or from mecha/closet
+		return
+	if(has_gravity(user))//You don't want use jet in gravity

```

Comment:
`!(L.is_flying() || L.is_floating()) || L.buckled`

Reply:
You know that is_floating() in space can be negated by magboots?

Reply:
```suggestion
	if(!(user.is_flying() || user.is_floating()) || user.buckled)//You don't want use jet in gravity or while buckled.
```